clients/drcachesim/tracer/raw2trace.h

```diff
@@ -876,6 +882,13 @@ template <typename T> class trace_converter_t {
             DR_CHECK(reinterpret_cast<trace_entry_t *>(buf) == buf_base,
                      "We shouldn't have buffered anything before calling "
                      "append_bb_entries");
+            if (saw_filter_endpoint && true) {
+                int file_type = impl()->get_file_type(tls);
+                file_type &= ~(OFFLINE_FILE_TYPE_FILTERED | OFFLINE_FILE_TYPE_IFILTERED |
+                               OFFLINE_FILE_TYPE_DFILTERED);
+                file_type |= OFFLINE_FILE_TYPE_DEFAULT;
+                impl()->set_file_type(tls, (offline_file_type_t)file_type);
```

Comment:
Where does this get used: the file type is already written for the current file, so this is for -split_windows?  But IIRC -split_windows requires the user to launch a separate instance of raw2trace for each file?  So I'm not understanding where this updated file type is used.

Reply:
These are the places where it was used:

analyze_elidable_addresses
```
    std::string
    analyze_elidable_addresses(void *tls, uint64 modidx, uint64 modoffs, app_pc start_pc,
                               uint instr_count)
    {
        int version = impl()->get_version(tls);
        // Old versions have no elision.
        if (version <= OFFLINE_FILE_VERSION_NO_ELISION)
            return "";
        // Filtered and instruction-only traces have no elision.
        if (TESTANY(OFFLINE_FILE_TYPE_FILTERED | OFFLINE_FILE_TYPE_NO_OPTIMIZATIONS |
                        OFFLINE_FILE_TYPE_INSTRUCTION_ONLY,
                    impl()->get_file_type(tls)))
            return "";

```

append_bb_entries
```
        bool instrs_are_separate =
            TESTANY(OFFLINE_FILE_TYPE_FILTERED | OFFLINE_FILE_TYPE_IFILTERED,
                    impl()->get_file_type(tls));
        bool is_instr_only_trace =
            TESTANY(OFFLINE_FILE_TYPE_INSTRUCTION_ONLY, impl()->get_file_type(tls));
```
append_memref
```
    std::string
    append_memref(void *tls, INOUT trace_entry_t **buf_in, const instr_summary_t *instr,
                  instr_summary_t::memref_summary_t memref, bool write,
                  std::unordered_map<reg_id_t, addr_t> &reg_vals,
                  OUT bool *reached_end_of_memrefs)
    {
        DR_ASSERT(
            !TESTANY(OFFLINE_FILE_TYPE_INSTRUCTION_ONLY, impl()->get_file_type(tls)));
        trace_entry_t *buf = *buf_in;
        const offline_entry_t *in_entry = nullptr;
        bool have_addr = false;
        bool have_type = false;
        reg_id_t base;
        int version = impl()->get_version(tls);
        if (memref.use_remembered_base) {
            DR_ASSERT(!TESTANY(OFFLINE_FILE_TYPE_FILTERED | OFFLINE_FILE_TYPE_IFILTERED |
                                   OFFLINE_FILE_TYPE_DFILTERED,
                               impl()->get_file_type(tls)));
```

Reply:
I see.  If you could add a comment here and one in the set_ routine explaining that this will not affect the written-out type but it does affect processing code for the rest of the trace as that is probably confusing to future readers too.

Reply:
Added comments.